### PR TITLE
Add support for building elasticsearch source on macos before monterey (Closes #87426)

### LINKED ISSUE #87426: Build source in Apple M1 fails if not using MacOs Monterey
### Elasticsearch Version

8.4

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Darwin kpl3x4j763.kalpatarugroup.com 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:06:51 PST 2021; root:xnu-7195.81.3~1/RELEASE_ARM64_T8101 arm64

### Problem Description

# Description
When running `./gradlew` in my Mac M1 with Big Sur I get an error from `sysctl` mentioning that `hw.perflevel0.physicalcpu` is an unknown property. When listing all properties using `sysctl -A` there's no `hw.perflevel*.*` properties at all.

Looking into this forum https://githubhot.com/repo/seanmonstar/num_cpus/issues/104 it seems the property is not displayed on versions before Monterey. I went ahead and changed this https://github.com/elastic/elasticsearch/blob/master/build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/info/ParallelDetector.java#L67-L69 to always use the `"hw.physicalcpu"` property and was able to get a clean build.

I think we should either specify somewhere that MacOS Monterey is required for building Elasticsearch in M1 macs or change the code in `ParallelDetector.java` such that it uses `"hw.physicalcpu"` for versions older than Monterey. I personally prefer the second option and I'll be happy to take the PR for fixing it if you all agree.

### Steps to Reproduce

N/A

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by jcbages**
Allow building elasticsearch source code from macos versions older than Monterey by falling back to a previous `sysctl` variable when determining the parallelism of the build

Tested it in my Mac running Big Sur successfully

Closes https://github.com/elastic/elasticsearch/issues/87426

### CODE REVIEW

File: build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/info/ParallelDetector.java [hunk 1]
```diff
@@ -90,4 +94,15 @@ private static boolean isMac(ProviderFactory providers) {
         return providers.systemProperty("os.name").getOrElse("").startsWith("Mac");
     }
 
+    private static boolean isMontereyOrNewer(ProviderFactory providers) {
+        String rawVersion = providers.systemProperty("os.version").getOrElse("").trim();
+        if (rawVersion.isEmpty()) {
+            Logging.getLogger(ParallelDetector.class).warn("Failed to validate MacOs version.");
```
Comment (mark-vieira) [inline build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/info/ParallelDetector.java:100]: Can we refactor the logger out into a static field? It's only used once here but it'll help us not duplicating this if we add more logging to the class down the road.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

**Comment by mark-vieira** [comment]
@elasticmachine ok to test

File: docs/changelog/87430.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+pr: 87430
```
Comment (mark-vieira) [inline docs/changelog/87430.yaml:1]: We can delete this changelog entry. This isn't a user-facing chnage so we typically don't include build changes in our release notes.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

File: build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/info/ParallelDetector.java (hunk 1 above)
Comment (jcbages) [reply build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/info/ParallelDetector.java]: Sure! I just added a final static variable called LOGGER with that. Let me know if it looks good to you!

File: docs/changelog/87430.yaml (hunk 2 above)
Comment (jcbages) [reply docs/changelog/87430.yaml]: Sounds good! I added it because of the checks below asking for it but I just erased it!

**Comment by jcbages** [comment]
@mark-vieira just addressed your changes and the build was successful!

**Review by mark-vieira — APPROVED** [review]

**Comment by mark-vieira** [comment]
Thanks for the pull request @jcbages!

**Pull request merged**